examples/incremental-ingest-from-db-example/README.md

```diff
@@ -0,0 +1,266 @@
+Overview
+--------
+
+In this example we will use the Versatile Data Kit to develop an incremental [ingestion](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/job_input.py#L90) [Data Job](https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job). This job will read data from one local SQLite database and write into another local SQLite database only the records that are not present in the target table, thus enriching the target.
+
+Before you continue, make sure you are familiar with the [Getting Started](https://github.com/vmware/versatile-data-kit/wiki/Getting-Started) section of the wiki.
+
+Code
+----
+
+The relevant Data Job code is available [here](https://github.com/vmware/versatile-data-kit/tree/main/examples).
+
+You can follow along and run this example Data Job on your machine to get first-hand experience working with Data Jobs; alternatively, you can use the available code as a template and extend it to make a Data Job that fits your use case more closely.
+
+Database
+--------
+
+We will be using a local SQLite database that was built for the purposes of the example. The initial version of this database can be found in the data job repository, subfolder "data".
+
+The contents of the uploaded data source are the following (**`incremental_ingest_example.db`** → table **`increm_ingest`**):
+
+| id | descr | reported_date |
+| --- | --- | --- |
+| 1 | record one | 10-01-2021 |
+| 2 | second record | 10-02-2021 |
+| 3 | this is record 3 | 10-03-2021 |
+
+After the initial ingestion takes place (the data job is run once, the target table is created and populated with the 3 records illustrated in the table above), the source table `increm_ingest` in the `incremental_ingest_example.db` database needs to be enriched with a few more records in order to illustrate how those new records will be incrementally ingested into the target database.
+
+The following records are added:
+
+| id | descr | reported_date |
+| --- | --- | --- |
+| 4 | that's a new record! | 11-01-2021 |
+| 5 | and another new record.. | 11-02-2021 |
+
+Please run the following Python code to insert the new records into the source table **only after the initial data job execution**:
+
+<details>
+  <summary>Add rows to source DB table</summary>
+
+```py
+import sqlite3
+
+# Create DB connection
```

Comment:
I made this code snippet VDK-independent on purpose since it only adds new rows to the source DB table for the purposes of illustrating how they would be incrementally ingested into the target table. This code snippet is not part of the incremental ingestion data job, that's why it's also not wrapped in a `run` function. Do you think this should be changed?

Reply:
can we simplify, just provide a link to the sqlite db file with the data pre-populated?  And users can download it.

Reply:
Addressed in the latest commit. Now the enriched data source is added to a separate folder and the users can directly download it.